### ISSUE TITLE
Hyper-parameters to replicate baseline results on COVID-FL?

### ISSUE THREAD

**Issue by siomvas** [rui-yan/SSL-FL#21]
Hello, I am trying to establish some baselines on COVID-FL.

My code-base is unfortunately quite different, so I am mostly going by what's written in the paper in trying to pinpoint hyper-parameter settings to achieve the following reported results:

Random init: 91.42 central accuracy and 77 on the split
Imagenet pre-trained: 95.35 central accuracy and 86.89 on the split. 

AdamW with: starting lr=3e-3, wd=0.05, decayed with cosine decay
Augmentations:  RandomResizedCrop((224, 224), scale=(0.8, 1.2)), RandomRotation(degrees=10), transforms.RandomHorizontalFlip()

Going through the repo it seems there's a number of other tricks included, 
- layer decay
- Random erasing
- mixup
- cutmix
- lr seems to be 3e-3 * 64/256

Can you point me in the right direction in terms of what's necessary to replicate your results?

At the moment, fine-tuning the torchvision imagenet pre-trained ViT on COVIDFL for 100 rounds I get 43.7% accuracy.

**Comment by rui-yan** [comment]
Hi, Is the unsatisfactory accuracy specific to ImageNet pre-trained ViT, or does it persist with other methods? The 43.7% accuracy is unexpectedly low and it seems unlikely that hyperparameters alone could account for such a drop. 

Could you check the integrity of your data loaders, ensuring that both training and test images are being loaded accurately? 

Try apply this preprocessing on the test data for fine-tuned model evaluation:
```transform = transforms.Compose([
                transforms.Resize(size=args.input_size),
                transforms.CenterCrop(size=(args.input_size, args.input_size)), 
                transforms.ToTensor(),
                transforms.Normalize(
                    mean=torch.tensor(mean),
                    std=torch.tensor(std))
                ])
```
Check the code here to confirm that the data loaders are configured correctly: https://github.com/rui-yan/SSL-FL/blob/main/code/util/datasets.py
